core/dbt/graph/selector_methods.py

```diff
@@ -478,42 +478,26 @@ def check_macros_modified(self, node):
             previous_macros = []
             return self.recursively_check_macros_modified(node, previous_macros)
 
-    def check_modified(self, old: Optional[SelectorTarget], new: SelectorTarget) -> bool:
+    # TODO check modifed_content and check_modified macro seems a bit redundent
```

Comment:
@gshank  do you know what's the main difference between these two functions? seems a little bit confusing here

Reply:
I think the way you're doing this is fine. The `check_modified` method (`state:modified` for users) is meant to be the superset of all checks. Renaming this so it clearly maps to `same_contents` (the umbrella check) makes sense